src/main/java/ua/team3/carsharingservice/service/impl/RentalServiceImpl.java

```diff
@@ -128,6 +130,24 @@ private void validateRentalPeriod(RentalRequestDto dto) {
 
     private void validateRentalPermissionFor(User user) {
         checkUnreturnedCarsFor(user);
+        ensureUserHasNoDebt(user);
+    }
+
+    private void ensureUserHasNoDebt(User user) {
+        List<Rental.Status> debtStatuses = Arrays.asList(
+                Rental.Status.PENDING,
+                Rental.Status.OVERDUE
+        );
+        List<Rental> rentalsWithDebt = rentalRepository.findByStatusIn(debtStatuses);
+        if (!rentalsWithDebt.isEmpty()) {
+            String rentalDates = rentalsWithDebt.stream()
+                    .map(r -> r.getRentalDate().toString())
+                    .collect(Collectors.joining(", "));
+            throw new ForbiddenRentalCreationException(
+                    "The user has debt the rental with the following rental dates: "
+                            + rentalDates
```

Comment:
Maybe it would be better to pass the id? So that the client can see which ones he needs to pay for